R/data-storage-log-file.R

```diff
@@ -0,0 +1,185 @@
+#' Data storage class for JSON Log File
+#'
+#' @description
+#' Implementation of the DataStorage R6 class to a JSON log file backend using a unified
+#' API for read/write operations
+#'
+#' @export
+#'
+#' @examples
+#' data_storage <- DataStorageLogFile$new(
+#'   username = "test_user",
+#'   log_file_path = tempfile(pattern = "user_stats", fileext = ".json"),
+#'   session_file_path = tempfile(pattern = "session_details", fileext = ".json")
```

Comment:
Should we call it a JSON file if it really isn't a valid file of that format? maybe rename it to `.txt` or `.log` would be better

Even if it's just an example

Reply:
Agree on close connection. I realized `close()` can just return nothing and not worry about `close_connection()`. Thank you for pointing that out.

I updated the file extension to `.txt`.